categorize_adjectives/hierarchical.py

```diff
@@ -0,0 +1,92 @@
+"""Clustering the adjectives in a hierarchical manner"""
+import os
+import pickle
+
+def cluster_cluster_distance(cluster_1, cluster_2):
+    """Calculate distance between clusters
+
+    It is defined as average over distance between adj1, adj2
+    pairs where adj1 belongs to cluster_1 and adj2 belongs to cluster_2
+
+    returns distance between the two clusters
+    """
+    average_distance = 0
+    for adj_1 in cluster_1:
+        for adj_2 in cluster_2:
+            average_distance += distance_metric[(adj_1, adj_2)]
```

Comment:
how does this compile? how is distance_metric accessed as its not a global variable?

Reply:
because it is declared in main, it acts as a global variable